@Perryo Getting this error right now:

`cv2.error: /tmp/opencv20150906-42178-3d0iam/opencv-2.4.12/modules/imgproc/src/histogram.cpp:192: error: (-215) ranges[i] && ranges[i][0] < ranges[i][1] in function histPrepareImages`

In the call for calculating the histogram. Investigating now.

--- a/Project 3/block_quantize.py
+++ b/Project 3/block_quantize.py
@@ -72,7 +72,6 @@ def quantize(frame_data, n):
 
     result = list()
     frame_block_dict = {}   #stores block dictionary which contains histogram data for blocks
-    block_hist_dict = {}    #stores histogram data for blocks of a frame
     frame_num = 0
 
     #for each entry in frame_data
@@ -80,6 +79,7 @@ def quantize(frame_data, n):
         frame_num += 1
         print 'Processing regions of frame: ' + str(frame_num)
         for block_x in range(0, len(frame), 8):
+            block_hist_dict = {}    #stores histogram data for blocks of a frame
             for block_y in range(0, len(frame[block_x]), 8):
                 block = frame[block_x:block_x+8, block_y:block_y+8]
                 block_dict,block_hist = quantize_block(block, n, frame_num, block_x, block_y)

--- a/Project 3/block_quantize.py
+++ b/Project 3/block_quantize.py
@@ -51,6 +51,9 @@ def quantize_block(block, bins, frame_num, block_x, block_y):
 
     result = list()
 
+    if min_value == max_value:
+        max_value += 1
+
     block_hist = cv2.calcHist([block.astype(np.uint8)],[0],None,[bins],[min_value,max_value]) #We could do our local min/max here but lets keep the range (0,256) the same for comparison reasons
 
     for key in frame_occurances:

--- a/Project 3/match_frame.py
+++ b/Project 3/match_frame.py
@@ -91,12 +91,11 @@ def show_ten_quantized_closest(frame_data,frame_block_dict,target_frame_number, 
             frame_score = float(0)
             for keyB in frame_block_dict[keyA]:
                 block_hist = frame_block_dict[keyA][keyB[0],keyB[1]]
-                frame_score += cv2.compareHist(target_frame_block_hist_dict[keyB[0],keyB[1]],block_hist, 1) #Chi-Square compare method, smaller number = higher similarity
+                frame_score += cv2.compareHist(target_frame_block_hist_dict[keyB[0],keyB[1]],block_hist, 0) #Correlation compare
 
         top_ten_frames.append((keyA, frame_score))
 
     top_ten_frames.sort(key=lambda tup: tup[1])  # sorts in place
-    top_ten_frames = top_ten_frames[:10]
 
     top_ten_frames = list((x[0] for x in top_ten_frames))    #just need to the frame number, not the diff so we strip that out
     for i in range(0,10):

